test/services/sorting/post_sorting_service_test.dart

```diff
@@ -0,0 +1,178 @@
+import "package:collection/collection.dart";
+import "package:flutter_test/flutter_test.dart";
+import "package:geolocator/geolocator.dart";
+import "package:proxima/models/database/post/post_firestore.dart";
+import "package:proxima/services/sorting/post_sort_option.dart";
+import "package:proxima/services/sorting/post_sorting_service.dart";
+
+import "../../mocks/data/firestore_post.dart";
+import "../../mocks/data/geopoint.dart";
+
+void expectSorted(Iterable<double> iterable, bool ascending) {
+  final factor = ascending ? 1 : -1;
+  expect(iterable.isSorted((a, b) => factor * a.compareTo(b)), true);
+}
+
+void main() {
+  final sortingService = PostSortingService();
+
+  final postGenerator = FirestorePostGenerator();
+  final positions = GeoPointGenerator.generatePositions(userPosition0, 10, 0);
+  final posts = postGenerator.generatePostsAtDifferentLocations(positions);
```

Comment:
Very nice job with the testing ! 
I have little suggestion, I think we should declare those kind of variables as `late` that we initalize in the `setUp` function. I believe this is the convention we adopted for the other tests and it ensures that each test has those variables freshly created when running and so that they were not modified by other tests.

I also wonder if we could add a "global" group to encapsulate all the testing.
I think we could have something like:

```suggestion
  group("Testing post sorting service", () {
    late PostSortingService sortingService;

    late FirestorePostGenerator postGenerator;
    late List<GeoPoint> positions;
    late List<PostFirestore> posts;

    setUp(() {
      sortingService = PostSortingService();

      postGenerator = FirestorePostGenerator();
      positions = GeoPointGenerator.generatePositions(userPosition0, 10, 0);
      posts = postGenerator.generatePostsAtDifferentLocations(positions);
    });
    
    // All the other groups that are performing tests
    // ...
    
  }
```

Note: if you go with this suggestion, I think you'll need to also implement the next suggestion otherwise I believe you'll get an error because some variable won't be initialized.

Reply:
I have a question and a remark on this suggestion:
- Does this really make sense to put all my tests in the same group? This really feels like using groups just for using groups, without having them giving real quality to me at first sight.
- My idea was that it was a lot of boilerplate to declare my variables as `late` and define them in the setup; since they never change. However, on reflection, this appears to be a good way to avoid future mistakes and non-deterministic tests. I'll implement this.

Reply:
- I don't know if it is strictly necessary but I quite like the idea having a big group that encapsulates everything and that tells what service we are testing. And then we have smaller groups to tell which part of the service we are testing. But this is only a personal opinion.
- Nice thank you :+1:

Reply:
- If you're ok with this, I will not implement this suggestion. I think test names are already long enough, and that this is not strictly necessary.
- Fixed in d6c9c4ecd56bf2b95cdcdb3926ee37184c6624b8.

Reply:
I understand, it's alright for me :+1: